STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+# Front-end style guide
+
+This document outlines recommended code style, quirks, and best practices for front-end development on Kedro-Viz.
+
+<!-- @TODO table of contents -->
+
+## Code style and linting
+
+We use [Prettier](https://prettier.io/), [ESLint](https://eslint.org/), and [stylelint](https://stylelint.io/) to lint our SCSS and JavaScript. For the most part, we use the default config - you can find specific details in the `.prettierrc`, `.eslintrc.json`, and `.stylelintrc` config files. We recommend that you install the Prettier, ESLint, and stylelint plugins for your text-editor, and enable automatic formatting on save.
+
+Other than that, we mostly recommend following the [AirBnB JavaScript Style Guide](https://github.com/airbnb/javascript).
+
+##  Browser and device support
+
+We aim to support recent versions of major modern browsers - i.e. Chome, Firefox, Edge, Safari, and Opera, on both MacOS and Windows. Keeping progressive enhancement in mind, we aim for decent mobile support where possible, i.e. nothing should be broken on mobile, but it's okay for mobile/older browsers to have a slightly degraded experience.
+
+## Accessibility
+
+We aim to meet [WCAG 2.1 Level AA](https://www.w3.org/WAI/standards-guidelines/wcag/) standards where possible, while acknowledging that this is a data visualization, so some criteria will be difficult to pass without extraordinary effort. However we will do our best within reason.
```

Comment:
```suggestion
We aim to meet [WCAG 2.1 Level AA](https://www.w3.org/WAI/standards-guidelines/wcag/) standards where possible, while acknowledging that this is a data visualization tool, so some criteria will be difficult to pass without extraordinary effort. 
```

Reply:
I suggested removing the "However we will do our best within reason." as it's a bit defensive. Is there any particular guidance for tools that are visual, so we can claim to have consulted that?

Reply:
There is, and that's a good shout. The reason I wrote that is that I know there are other things that we can do that we're not currently doing, as it would be tricky and it's not a high priority at present. So I want to say something to the effect that we will try to make an effort to improve accessibility where reasonable.